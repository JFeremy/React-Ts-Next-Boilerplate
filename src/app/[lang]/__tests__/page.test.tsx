import { render } from '@testing-library/react';

import {
  getComponentsTranslation,
  getMetadataPageTranslation,
  getPageTranslation,
} from '@/utils/translate';

import Home, { generateMetadata, generateStaticParams } from '../page';

jest.mock('@/utils/translate', () => ({
  getPageTranslation: jest.fn(),
  getComponentsTranslation: jest.fn(),
  getMetadataPageTranslation: jest.fn(),
}));

describe('Home', () => {
  describe('Page', () => {
    it('page matches snapshot', async () => {
      const mockPTranslationData = {
        server: 'Necessitatibus facere odio',
      };
      const mockCTranslationData = {
        hello: {
          title: 'Lorem, ipsum dolor.',
        },
        header: {
          title: 'sit amet consectetur adipisicing elit.',
          localeSwitcher: 'Lorem ipsum dolor',
        },
        counter: {
          more: 'quibusdam',
          less: 'sed',
        },
      };

      (getPageTranslation as jest.Mock).mockReturnValue(mockPTranslationData);
      (getComponentsTranslation as jest.Mock).mockReturnValue(
        mockCTranslationData,
      );

      const view = render(<Home params={{ lang: 'fr' }} />);

      expect(view.container).toMatchSnapshot();
    });
  });

  describe('generateStaticParams', () => {
    it('should generate static params for all available locales', async () => {
      const staticParams = await generateStaticParams();
      expect(staticParams).toHaveLength(2);
      expect(staticParams).toEqual(
        expect.arrayContaining([{ lang: 'en' }, { lang: 'fr' }]),
      );
    });
  });

  describe('generateMetadata', () => {
    it('should generate metadata for locale', async () => {
      const mockMTranslationData = {
        title: 'Lorem ipsum dolor sit amet.',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Illo architecto mollitia blanditiis, accusantium quibusdam possimus officia eligendi eveniet nostrum iure.',
      };

      (getMetadataPageTranslation as jest.Mock).mockReturnValue(
        mockMTranslationData,
      );

      const metadata = await generateMetadata({ params: { lang: 'en' } });
      expect(metadata).toEqual(mockMTranslationData);
      expect(getMetadataPageTranslation as jest.Mock).toHaveBeenCalledWith(
        'en',
        'home',
      );
    });
  });
});
