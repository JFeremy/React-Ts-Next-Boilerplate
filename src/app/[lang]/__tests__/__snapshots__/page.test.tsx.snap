// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page page matches snapshot 1`] = `
<div>
  <div
    class="pos_relative max-w_8xl mx_auto px_4 md:px_6 lg:px_8 p_6"
  >
    <div
      class="d_flex items_center justify_center gap_5 flex_column"
    >
      <div
        class="font_anybody"
      >
        <p>
          Lorem ipsum dolor
        </p>
        <ul>
          <li>
            <a
              href="/en"
            >
              en
            </a>
          </li>
          <li>
            <a
              href="/fr"
            >
              fr
            </a>
          </li>
        </ul>
      </div>
      <p>
        sit amet consectetur adipisicing elit.
      </p>
      <p>
        Docs
      </p>
      <h2>
        fr
      </h2>
      <div
        class="d_flex items_center justify_center bg_red.200"
      >
        <div
          class="fs_2xl font_bold font_fira"
        >
          Lorem, ipsum dolor.
        </div>
      </div>
      <p>
        Necessitatibus facere odio
      </p>
      <div
        class="d_flex items_center justify_center gap_5 flex_row"
      >
        <button
          class="text_red text_center bg_gray.800 rounded_2 fs_1rem px_10px py_5px"
        >
          sed
        </button>
        <p>
          0
        </p>
        <button
          class="text_red text_center bg_gray.800 rounded_2 fs_1rem px_10px py_5px"
        >
          quibusdam
        </button>
      </div>
    </div>
  </div>
</div>
`;
